src/main/java/com/tsmms/skoop/user/command/UserGlobalPermissionCommandController.java

```diff
@@ -0,0 +1,67 @@
+package com.tsmms.skoop.user.command;
+
+import com.tsmms.skoop.user.command.ReplaceUserGlobalPermissionListCommand.UserGlobalPermissionEntry;
+import com.tsmms.skoop.user.GlobalPermissionRequest;
+import com.tsmms.skoop.user.GlobalPermissionResponse;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toSet;
+import static java.util.stream.Collectors.toList;
+
+@Api(tags = "UserGlobalPermissions")
+@RestController
+public class UserGlobalPermissionCommandController {
```

Comment:
I really like that you took into account that we are dealing with permissions granted to users. Because just "global permission" could also mean "what is the authenticated user allowed to do in the application".

I would just suggest to rename it to `GlobalUserPermissionCommandController` for better readability. The same applies to the node entity and the service bean.

Reply:
I have just changed as you suggested.

I have been thinking for a while whether to use `User` everywhere or only in controllers to emphasise the places in source code dealing with relationships. The both options seemed to be very similar.

And the same thing with suffixing / prefixing with `User`. But now I see as soon as we have `UserPermission` it is better to have `GlobalUserPermission` than `UserGlobalPermission`.

Thank you 👍